agent/config.go

```diff
@@ -1966,12 +1977,23 @@ func MergeConfig(a, b *Config) *Config {
 		result.SessionTTLMin = b.SessionTTLMin
 		result.SessionTTLMinRaw = b.SessionTTLMinRaw
 	}
-	if len(b.HTTPAPIResponseHeaders) != 0 {
-		if result.HTTPAPIResponseHeaders == nil {
-			result.HTTPAPIResponseHeaders = make(map[string]string)
+
```

Comment:
Can you move that functionality into `DecodeConfig`, please? We're doing the other arg mangling there as well.